src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -382,6 +382,11 @@ public void OnStarting(Func<object, Task> callback, object state)
         {
             lock (_onStartingSync)
             {
+                if (HasResponseStarted)
+                {
+                    ThrowResponseAlreadyStartedException(nameof(OnStarting));
```

Comment:
So with the string it inserts a load string call then calls the function
```asm
mov      ecx, 0x498F
call     CORINFO_HELP_STRCNS_CURRENT_MODULE
mov      rcx, rax
call     Frame:ThrowResponseAlreadyStartedException(ref)
```
With the enum it just sets a value
```asm
mov      ecx, 3
call     Frame:ThrowResponseAlreadyStartedException(int)
```
It does push the load string off to `Frame:ThrowResponseAlreadyStartedException` as the string still needs to get loaded; but that does really get called, so the the main function is more trim and the instruction cache hotter.